zanata-client-commands/src/main/java/org/zanata/client/commands/push/RawPushCommand.java

```diff
@@ -75,10 +96,174 @@ public RawPushCommand(PushOptions opts, RestClientFactory clientFactory) {
         client = getClientFactory().getFileResourceClient();
     }
 
+    /**
+     * Extract extensions in square bracket
+     * input pattern: [extension;extension]
+     */
+    private List<String> extractExtensions(String rawExtensions) {
+        if(StringUtils.isEmpty(rawExtensions)) {
+            return Collections.emptyList();
+        }
+
+        //remove square bracket in first and last character
+        rawExtensions = rawExtensions.substring(1, rawExtensions.length()-1);
+        return Arrays.asList(rawExtensions.split(";"));
+    }
+
+    /**
+     * Extract input string to type and extensions
+     * input pattern: fileType[extension;extension], [extension;extension]
+     *
+     * @param typeWithExtension
+     * @return result[0] - type, result[1] - extensions. e.g [txt,html]
+     */
+    private String[] extractTypeAndExtension(String typeWithExtension) {
```

Comment:
- [ ] I was suggesting to use a single regex, but I think it is better to keep the two methods to get each of the parts (just use the same regex in them).

Java is too clunky to return multiple results like this without a class for the results. I was going to say it is not worth making a class just for this, but then it might make for cleaner code to have a class that represents 1 of the file type entries (type + extensions) - there are several other places such a class could be used.